<li id="rule_RSPEC-3251" class="rule filter_Code_Smell_Detection ">
      <div class="rule-title"><input type="checkbox" id="toggle_RSPEC-3251" class="rule-toggle" value="RSPEC-3251"></input>
        <label for="toggle_RSPEC-3251" class="toggle" id="label_RSPEC-3251">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 249 253.5"><path d="M93.2 206c-1.9 0-3.8-.7-5.3-2.2-2.9-2.9-2.9-7.7 0-10.6l67.2-67.2L87 58c-2.9-2.9-2.9-7.7 0-10.6 2.9-2.9 7.7-2.9 10.6 0l73.3 73.3c1.4 1.4 2.2 3.3 2.2 5.3s-.8 3.9-2.2 5.3l-72.5 72.5c-1.4 1.5-3.3 2.2-5.2 2.2z"/></svg>
          Implementations should be provided for &quot;partial&quot; methods
          <a href="#RSPEC-3251"><?xml version="1.0" encoding="iso-8859-1"?>
<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 54.971 54.971" style="enable-background:new 0 0 54.971 54.971;" xml:space="preserve">
<g>
	<path d="M51.173,3.801c-5.068-5.068-13.315-5.066-18.384,0l-9.192,9.192c-0.781,0.781-0.781,2.047,0,2.828
		c0.781,0.781,2.047,0.781,2.828,0l9.192-9.192c1.691-1.69,3.951-2.622,6.363-2.622c2.413,0,4.673,0.932,6.364,2.623
		s2.623,3.951,2.623,6.364c0,2.412-0.932,4.672-2.623,6.363L36.325,31.379c-3.51,3.508-9.219,3.508-12.729,0
		c-0.781-0.781-2.047-0.781-2.828,0s-0.781,2.048,0,2.828c2.534,2.534,5.863,3.801,9.192,3.801s6.658-1.267,9.192-3.801
		l12.021-12.021c2.447-2.446,3.795-5.711,3.795-9.192C54.968,9.512,53.62,6.248,51.173,3.801z"/>
	<path d="M27.132,40.57l-7.778,7.778c-1.691,1.691-3.951,2.623-6.364,2.623c-2.412,0-4.673-0.932-6.364-2.623
		c-3.509-3.509-3.509-9.219,0-12.728L17.94,24.306c1.691-1.69,3.951-2.622,6.364-2.622c2.412,0,4.672,0.932,6.363,2.622
		c0.781,0.781,2.047,0.781,2.828,0s0.781-2.047,0-2.828c-5.067-5.067-13.314-5.068-18.384,0L3.797,32.793
		c-2.446,2.446-3.794,5.711-3.794,9.192c0,3.48,1.348,6.745,3.795,9.191c2.446,2.447,5.711,3.795,9.191,3.795
		c3.481,0,6.746-1.348,9.192-3.795l7.778-7.778c0.781-0.781,0.781-2.047,0-2.828S27.913,39.789,27.132,40.57z"/>
</g>
</svg>
</a>
        </label>
        <div class="rule-description">
          <p><tt>partial</tt> methods allow an increased degree of flexibility in programming a system. Hooks can be added to generated code by invoking methods that define their signature, but might not have an implementation yet. But if the implementation is still missing when the code makes it to production, the compiler silently removes the call. In the best case scenario, such calls simply represent cruft, but in they worst case they are critical, missing functionality, the loss of which will lead to unexpected results at runtime.</p>

<p>This rule raises an issue for partial methods for which no implementation can be found in the assembly.</p>

<h2><a name="NoncompliantCodeExample"></a>Noncompliant Code Example</h2>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
partial class C
{
  partial void M(); <span class="code-comment">//Noncompliant
</span>
  void OtherM()
  {
    M(); <span class="code-comment">//Noncompliant. Will be removed.
</span>  }
}
</pre>
</div></div>
        </div>
      </div>
    </li>